src/db.c

```diff
@@ -2170,20 +2167,67 @@ int sortGetKeys(struct redisCommand *cmd, robj **argv, int argc, getKeysResult *
                 i += skiplist[j].skip;
                 break;
             } else if (!strcasecmp(argv[i]->ptr,"store") && i+1 < argc) {
-                /* Note: we don't increment "num" here and continue the loop
+                /* Note: we want to keep update the same store key reference
                  * to be sure to process the *last* "STORE" option if multiple
                  * ones are provided. This is same behavior as SORT. */
-                found_store = 1;
-                keys[num].pos = i+1; /* <store-key> */
-                keys[num].flags = CMD_KEY_OW | CMD_KEY_UPDATE;
-                break;
+                if (!store_key)
+                    store_key = &keys[num++];
+                store_key->pos = i+1; /* <store-key> */
+                store_key->flags = CMD_KEY_OW | CMD_KEY_UPDATE;
+                i++;
+            } else if ( (!strcasecmp(argv[i]->ptr,"get") || !strcasecmp(argv[i]->ptr,"by")) && i+1 < argc) {
+                /* '#' does not stands for real key, but is just being replaced with the list content in get. */
+                if (with_patterns && strcasecmp(argv[i+1]->ptr,"#")) {
+                    /* Note: we want to keep update the same pattern reference
+                     * to be sure to process the *last* "GET" or "BY" options.
+                     * if multiple ones are provided we will only return the first one.
+                     * This is since we only use these pattern to validate ACL rules which are currently only
+                     * looking for validating that in case at-least one pattern exists - the selector
+                     * should allow full key access
```

Comment:
```suggestion
                     * This is since we only use these pattern to validate ACL rules which currently require
                     * full read key access if a pattern exists.
```